src/core/sio1.cc

```diff
@@ -21,127 +21,139 @@
 
 void PCSX::SIO1::interrupt() {
     SIO1_LOG("SIO1 Interrupt (CP0.Status = %x)\n", PCSX::g_emulator->m_cpu->m_regs.CP0.n.Status);
-    m_statusReg |= SR_IRQ;
-    psxHu32ref(0x1070) |= SWAP_LEu32(0x100);
-    if (!m_slices.m_sliceQueue.empty() && m_slices.getBytesRemaining() > 1) scheduleInterrupt(SIO1_CYCLES);
+    m_regs.status |= SR_IRQ;
+    psxHu32ref(0x1070) |= SWAP_LEu32(IRQ8_SIO);
+    if (m_fifo.size() > 1) scheduleInterrupt(SIO1_CYCLES);
 }
 
 uint8_t PCSX::SIO1::readData8() {
-    const uint8_t ret = m_slices.getByte();
     updateStat();
-    psxHu8(0x1050) = ret;
+    if (m_regs.status & SR_RXRDY) {
+        m_regs.data = m_fifo.byte();
+        psxHu8(0x1050) = m_regs.data;
+    }
+    updateStat();
 
-    return ret;
+    return m_regs.data;
 }
 
 uint8_t PCSX::SIO1::readStat8() {
     updateStat();
-    return m_statusReg & 0xFF;
+    return m_regs.status & 0xFF;
```

Comment:
The return type will naturally cull this, no need for the `&&`

```suggestion
    return m_regs.status;
```